eth/rewardservice.go

```diff
@@ -54,9 +56,16 @@ func (s *RewardService) Start(ctx context.Context) error {
 			}
 		case <-roundSink:
 			go func() {
-				err := s.tryReward()
+				tx, err := s.tryReward()
 				if err != nil {
 					glog.Errorf("Error trying to call reward err=%q", err)
+					var txHash string
+					if tx != nil && !strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
Could you move "parsing" the error to `monitor.RewardCallError()`? I think you can pass there the whole `tx` and `err`. I'd try to avoid parsing the error in multiple places.